tools/scripts/broker_test_consumer.sh

```diff
@@ -0,0 +1,133 @@
+
+# variable to store jms destination topic/queue.Default value is queue
+jmx_file_location="test_plan/broker_test_queue_consumer.jmx"
+destination=""
+connection_factory_name=""
+jndi_file_location=""
+
+# get inputs from the user -p <location of the properties file> -d topic/queue
+while getopts "hp:d:t:h:v" OPTION
+do
+     case $OPTION in
+         p)
+            properties_file_location=$OPTARG
+            break
+            ;;
+         d)
+            case $OPTARG in
+                queue)
+                    destination="QueueName"
+                    connection_factory_name="QueueConnectionFactory"
+                    jndi_file_location="resources/jndi_queue.properties"
+                    jmx_file_location="test_plan/broker_test_queue_consumer.jmx"
+                ;;
+                topic)
+                    destination="TopicName"
+                    connection_factory_name="TopicConnectionFactory"
+                    jndi_file_location="resources/jndi_topic.properties"
+                    jmx_file_location="test_plan/broker_test_topic_consumer.jmx"
+                    ;;
+                ?)
+                echo $OPTARG
+                    echo "$OPTARG is an invalid destination.JMS destination should be a queue or a topic"
+                    exit
+                ;;
+            esac
+            break
+            ;;
+         h)
+            help_text="Welcome to ballerina message broker micro-benchmark tool\n\nUsage:\n\t./broker_test_consumer.sh [command].\n\nCommands\n\t-h  ask for help\n\t-p  set the location of the properties file\n\t-d  set jms destination type queue/topic\n"
+            printf "$help_text"
+            exit
+            ;;
+         ?)
+            printf "Invalid command.Run ./broker_test_consumer.sh -h for usage.\n"
+            exit
+            ;;
+     esac
+done
+
+# check for target folder
+if [ ! -e target/ ];
+then
+    mkdir -p target/subscriber
+fi
+
+# hash-map to store user desired parameters
+declare -A user_inputs=(["jmeter_home"]="" ["thread_count"]="1" ["number_of_messages"]="1000000")
+
+# Method to extract values from property file
+getProperty()
+{
+    local property_key=$1
+    local user_value=`cat ${properties_file_location} | grep ${property_key} | cut -d'=' -f2`
+    echo "$user_value"
+}
+
+for parameter in "${!user_inputs[@]}";
+do
+    value=$(getProperty $parameter)
+    if [ "$value" != '' ]
+        then
+            user_inputs["$parameter"]="$value"
+    fi
+done
+
+# validate inputs
+if [ ${user_inputs["thread_count"]} == '0' ]
+    then
+        # Command invoked cannot execute
+        echo Thread count cannot be zero
+        exit
+    fi
+
+# Summarizing inputs
+echo Jmeter home is set to - ${user_inputs["jmeter_home"]}
+echo Starting test process with a throughput - ${user_inputs["throughput"]}
+
+# calculate loop count
+loop_count=$(echo "${user_inputs["number_of_messages"]}/${user_inputs["thread_count"]}+1" | bc)
```

Comment:
why ${user_inputs["thread_count"]}**+1**

Reply:
This was added for some test purposes and forgot to remove.Now removed.